patches/server/0958-Add-Player-sendEquipmentChange-Map-API.patch

```diff
@@ -0,0 +1,39 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Aya <31237389+tal5@users.noreply.github.com>
+Date: Fri, 20 Jan 2023 13:49:59 +0000
+Subject: [PATCH] Add Player#sendEquipmentChange(Map) API
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index b32f44beab2c9790ee2da8403e362e8b3ecc6175..3043da912195d468baa2a0b7b995169e003a7760 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -1055,20 +1055,22 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+         this.sendSignChange0(components, loc, dyeColor, hasGlowingText); // Paper
+     }
+ 
++    // Paper start
+     @Override
+-    public void sendEquipmentChange(LivingEntity entity, EquipmentSlot slot, ItemStack item) {
++    public void sendEquipmentChange(LivingEntity entity, Map<EquipmentSlot, ItemStack> equipmentChanges) {
+         Preconditions.checkArgument(entity != null, "entity must not be null");
+-        Preconditions.checkArgument(slot != null, "slot must not be null");
+-        Preconditions.checkArgument(item != null, "item must not be null");
++        Preconditions.checkNotNull(equipmentChanges, "equipmentChanges must not be null");
+ 
+         if (this.getHandle().connection == null) return;
+ 
+-        List<Pair<net.minecraft.world.entity.EquipmentSlot, net.minecraft.world.item.ItemStack>> equipment = Arrays.asList(
+-                new Pair<>(CraftEquipmentSlot.getNMS(slot), CraftItemStack.asNMSCopy(item))
+-        );
++        List<Pair<net.minecraft.world.entity.EquipmentSlot, net.minecraft.world.item.ItemStack>> equipment = new ArrayList<>(equipmentChanges.size());
++        for (Map.Entry<EquipmentSlot, ItemStack> entry : equipmentChanges.entrySet()) {
++            equipment.add(new Pair<>(CraftEquipmentSlot.getNMS(entry.getKey()), CraftItemStack.asNMSCopy(entry.getValue())));
++        }
+ 
+         this.getHandle().connection.send(new ClientboundSetEquipmentPacket(entity.getEntityId(), equipment));
+     }
++    // Paper end
```

Comment:
Move this up to the actual end of the diff. (right below the changed "equipment" list creation.